### DISCUSSION TITLE
Jinja2: detect if user is logged in

### DISCUSSION THREAD

**Discussion by bert2002** [fastapi-users/fastapi-users#764] [Q&A] [answered]
Hi,
is there a best practice on how to detect if a user is authenticated in jinja2 templates for pages? I tried, e.g.:

```
@app.get("/", response_class=HTMLResponse)
def website(request: Request, User = Depends(current_active_verified_user)):
  return templates.TemplateResponse("index.html", {"request": request})
```
This does only work when the user is logged in, but not to detect the status of a user.

Wondering what is the most secure solution :)

bert

**Comment by frankie567** [comment, marked as answer]
When you call `current_active_verified_user` and the user is not logged in, FastAPI Users will automatically raise a 401 error. It means that the user *must* be authenticated to access this route.

If you wish to know if the user is logged in or not, you can use the `optional` argument on the `current_user` dependency factory (https://fastapi-users.github.io/fastapi-users/usage/current-user/#current_user).

```py
optional_current_active_verified_user = fastapi_users.current_user(active=True, verified=True, optional=True)
```

In this case, you'll get `None` if the user is not authenticated, but no error will be raised. You can pass the `user` object to your Jinja context and use it inside the template.

```py
@app.get("/", response_class=HTMLResponse)
def website(request: Request, user = Depends(optional_current_active_verified_user)):
  return templates.TemplateResponse("index.html", {"request": request, "user": user})
```
